File: test_archives/py_tests/batch2/test_function7.py
from functions.function7 import function7
import pytest

def test_function7_1():
    assert function7(1, 2.0, True, 'test', [1, 2, 3], {'a':1, 'b':2}) == 13

def test_function7_2():
    assert function7(2, 3.3, False, '', [], {}) == 5.3

File: test_archives/cpp_tests/batch2/test_function7.cpp
#include "../catch.hpp"
#include "../functions/function7.cpp"

TEST_CASE( "Test function7_1", "[function7_1]" ) {
    REQUIRE(function7(1, 2.0, true, "test", {1, 2, 3}, {{"a", 1}, {"b", 2}}) == 13);
}

TEST_CASE( "Test function7_2", "[function7_2]" ) {
    REQUIRE(function7(2, 3.3, false, "", {}, {}) == 5.3);
}